src/factor_graph.h

```diff
@@ -175,8 +173,17 @@ class CompactFactorGraph {
 };
 
 inline double CompactFactorGraph::potential(
-    const CompactFactor& factor, const variable_value_t assignments[]) {
-  return factor.potential(vifs.get(), assignments, -1, -1);
+    const CompactFactor& factor, const variable_value_t assignments[],
+    factor_value_t factor_value) {
+  // For boolean, this is stored in CompactFactor.value;
+  //     caller shouldn't set the factor_value arg.
+  // For categorical, this is stored in Factor.factor_params.value.
+  //     caller should pass in via the factor_value arg.
+  // TODO: better data structures than the factor_params duct tape...
+  if (factor_value == Factor::DEFAULT_VALUE) {
+    factor_value = factor.value;
+  }
```

Comment:
This is not so...factor and feature are distinct...

On Sun, Jul 17, 2016, 10:01 Jaeho Shin <notifications@github.com> wrote:

> In src/factor_graph.h
> <https://github.com/HazyResearch/sampler/pull/42#discussion_r71085609>:
>
> > @@ -175,8 +173,17 @@ class CompactFactorGraph {
> >  };
> >
> >  inline double CompactFactorGraph::potential(
> > -    const CompactFactor& factor, const variable_value_t assignments[]) {
> > -  return factor.potential(vifs.get(), assignments, -1, -1);
> > +    const CompactFactor& factor, const variable_value_t assignments[],
> > +    factor_value_t factor_value) {
> > +  // For boolean, this is stored in CompactFactor.value;
> > +  //     caller shouldn't set the factor_value arg.
> > +  // For categorical, this is stored in Factor.factor_params.value.
> > +  //     caller should pass in via the factor_value arg.
> > +  // TODO: better data structures than the factor_params duct tape...
> > +  if (factor_value == Factor::DEFAULT_VALUE) {
> > +    factor_value = factor.value;
> > +  }
>
> switch-casing on factor.func_id might be clearer? and perhaps the
> multiplication can be folded into CompactFactor::potential(_,_,_,_)?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/HazyResearch/sampler/pull/42/files/73792183d31eac97322ce3c5dd89223230078f47#r71085609>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHPtuK-4b8m0OIqmCYf6D_Q3uvYOuNdoks5qWl_SgaJpZM4JN8j9>
> .
>